src/parsers/perf/perfparser.cpp

```diff
@@ -606,6 +610,29 @@ struct PerfParserPrivate
          */
     }
 
+    void addSampleToSummary(const Sample& sample)
+    {
+        if (sample.time < applicationStartTime || applicationStartTime == 0) {
+            applicationStartTime = sample.time;
+        }
+        else if (sample.time > applicationEndTime || applicationEndTime == 0) {
+            applicationEndTime = sample.time;
+        }
+        if (!uniqueThreads.contains(sample.tid)) {
```

Comment:
once these are sets, simply do `uniqueThreads.insert(sample.tid)` - it will either insert when not exist, or error out. Doing a contains check first then inserting is going to be just as slow/fast, except when we actually insert - then we do the contains check twice.